### PR TITLE
Update jscs config

### PR DESCRIPTION
**Pull request by lukefrake**
Updated JSCS config to fit our current code guidelines.

For more info look [here](http://jscs.info/rules.html)

### CODE REVIEW

File: guidelines/shortbreaks.jscs.json [hunk 1]
```diff
@@ -1,26 +1,49 @@
...
+	"disallowSpaceAfterObjectKeys": true,
+	"disallowSpacesInCallExpression": true,
+	"disallowTrailingComma": true,
+	"disallowYodaConditions": true,
+	"requireCamelCaseOrUpperCaseIdentifiers": true,
+	"requireCapitalizedComments": true,
+	"requireCapitalizedConstructors": true,
+	"requireCommaBeforeLineBreak": true,
+	"requireSpaceBeforeBlockStatements": true,
+	"requireSpacesInsideObjectBrackets": "all",
```
Comment (kevinhodges) [inline guidelines/shortbreaks.jscs.json]: this is duplicated from line 19

Comment (lukefrake) [inline guidelines/shortbreaks.jscs.json]: Updated

**Comment by kevinhodges** [comment]
+1

**Comment by notABluesSinger** [comment]
Didn't think we minded yoda ?
```
"disallowYodaConditions": true,
```
not that I mind either way.

+1

**Comment by lukefrake** [comment]
@notABluesSinger - Not sure, I couldn't see it anywhere in our code base, and thought force the consistency as we don't do it anywhere else :)

**Comment by robhuzzey** [comment]
I thought we preferred yoda conditions? Or maybe that's just @hpoom that does? Maybe consistency is better.

**Comment by lukefrake** [comment]
Just to make sure we are on the same page, yoda conditions are this:

```
if ( value = var ) {
}
```

I've not seen anywhere in the code base with it on, but I can take it out if anybody thinks we should allow it?

**Comment by robhuzzey** [comment]
I just corrected your last comment because of a typo @lukefrake but ironically, that kind of typo is exactly what yoda conditions protect against:

you wrote:
if ( value = var ) {
}

**Comment by lukefrake** [comment]
hahaha :+1:

**Comment by hpoom** [comment]
Our coding guidelines do state we use yoda conditions. It probably came from me as I am huge fan of them.

Yes @lukefrake  it is:
`
if ( 1 == myVar ) {
}
`

The advantage being that if you only type one = you will get a compile error, whereas the other way round a single equal would just pass unnoticed.

**Comment by lukefrake** [comment]
Do we want to keep using the Yoda conditions then @hpoom?

Either it should go in here saying we shouldn't use it.
Or it should be in our guidelines saying we should.

Personally,  don't think it's as readable.

**Comment by hpoom** [comment]
Just get the team to agree on one and get it in the guidelines. I this was already in the guidelines.

**Comment by robhuzzey** [comment]
If it helps, I don't actually use them... I agree they are very useful, but the readability point is a valid one... we either make them standard so everybody does it by default, which may require some re-education, or we carry on as we are.

**Comment by lukefrake** [comment]
As long as everybody is okay with it, I'll leave this in here which means it fits what our current guidelines say and what our code looks like.

If anybody thinks this should be added in, do a PR on this repo adding it into the guidelines and all and then we can discuss on that repo what we want to do.

**Comment by robhuzzey** [comment]
+1

**Pull request merged**